File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
import math

def solution(n, words):
    answer = []
    temp = []
    
    for i in range(len(words)):
        if (i > 0 and words[i][0] != words[i - 1][len(words[i - 1]) -1]) or words[i] in temp:
            answer.append(i % n + 1)
            answer.append(math.floor(i / n) + 1)
            return answer
        
        temp.append(words[i])
        
    return [0,0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    var answer = [];
    let temp = [];
    
    for (let i = 0; i < words.length; i++) {
        if (i > 0 && words[i][0] !== words[i -1][words[i - 1].length - 1] || temp.includes(words[i])) {
            answer.push(i % n + 1);
            answer.push(Math.floor(i / n) + 1);
            return answer;
        }
        temp.push(words[i]);
    }

    return [0, 0];
}
